service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ReviewCMOControllerAboutToSubmitTest.java

```diff
@@ -117,22 +119,33 @@ void shouldSealPDFAndAddToSealedCMOsListWhenJudgeApprovesOrder() throws Exceptio
     }
 
     @Test
-    void shouldUpdateStateToIssueResolutionWhenNextHearingIssueResolutionAndCmoDecisionIsSendToAllParties() {
+    void shouldUpdateStateToIssueResolutionWhenNextHearingIsOfTypeIssueResolutionAndCmoDecisionIsSendToAllParties() {
         given(featureToggleService.isNewCaseStateModelEnabled()).willReturn(true);
 
         UUID cmoId = UUID.randomUUID();
-        CaseData caseData = buildCaseData(cmoId, SEND_TO_ALL_PARTIES);
+        CaseData caseData = buildCaseData(cmoId, SEND_TO_ALL_PARTIES, ISSUE_RESOLUTION);
         CaseData responseData = extractCaseData(postAboutToSubmitEvent(caseData));
 
         assertThat(State.ISSUE_RESOLUTION).isEqualTo(responseData.getState());
     }
 
+    @Test
+    void shouldUpdateStateToFinalHearingWhenNextHearingIssueIsOfTypeFinalAndCmoDecisionIsSendToAllParties() {
+        given(featureToggleService.isNewCaseStateModelEnabled()).willReturn(true);
+
+        UUID cmoId = UUID.randomUUID();
+        CaseData caseData = buildCaseData(cmoId, SEND_TO_ALL_PARTIES, FINAL);
+        CaseData responseData = extractCaseData(postAboutToSubmitEvent(caseData));
+
+        assertThat(State.FINAL_HEARING).isEqualTo(responseData.getState());
+    }
+
     @Test
     void shouldNotUpdateStateToIssueResolutionWhenFeatureToggledOff() {
         given(featureToggleService.isNewCaseStateModelEnabled()).willReturn(false);
 
```

Comment:
Do we need similar feature toggle test for Final Hearing state?

Reply:
Or maybe this test name could be changed to doNotUpdateStateWhenFeatureToggleIsOff to cater for both?

Reply:
let me know your thoughts :)

Reply:
hmm, what I think might be better is test - `shouldNotChangeStateWhenNewStateFeatureToggledIsToggledOff` and in that confirm both.

Reply:
Yep I like that idea, done.